File: 792-binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int first=0;
        int last=nums.size() -1;
        int mid=(first+last)/2;
        while(first<=last){
            mid=(first+last)/2;
            if(nums[mid]==target){
                return mid;
            }else if(nums[mid]<target){
                first=mid+1;
            }else{
                last=mid-1;
        }
    }
    return -1;
    }
};

File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length-1;
        while(left<=right){
            int mid = (left+right)/2;
            if(nums[mid]==target){
                return mid;
            }else if(target<nums[mid]){
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return -1;
    }
}